src/server/frontend_wayland/wayland_input_dispatcher.cpp

```diff
@@ -222,28 +222,34 @@ void mf::WaylandInputDispatcher::handle_pointer_motion_event(
     auto const axis_motion = std::make_pair(
         mir_pointer_event_axis_value(event, mir_pointer_axis_hscroll),
         mir_pointer_event_axis_value(event, mir_pointer_axis_vscroll));
-    bool const send_motion = (!last_pointer_position || position != last_pointer_position.value());
+    geom::Displacement const motion{
+        mir_pointer_event_axis_value(event, mir_pointer_axis_relative_x),
+        mir_pointer_event_axis_value(event, mir_pointer_axis_relative_y)};
+
+    bool const send_position = (!last_pointer_position || position != last_pointer_position.value());
     bool const send_axis = (axis_motion.first || axis_motion.second);
+    bool const send_motion = (motion != geom::Displacement{});
 
     last_pointer_position = position;
 
-    if (send_motion || send_axis)
```

Comment:
No need to drop this (probably it was added since you forked off). Obviously it will need to be updated to include `send_position`.